### ISSUE TITLE
iZettle SDK crashes the app, failing to process a `cardReaderBatteryStatusDidUpdate` call

### ISSUE THREAD

**Issue by leonid-s-usov** [iZettle/sdk-ios#115]
Please see the call stack here http://crashes.to/s/211b65d1e6d
We are using Reader One and the latest SDK 1.4.
Seems to be an issue with Low battery level. Plugging the iZettle device to a charger seems to stop the crashing

**Comment by mansbernhardt** [comment]
Thanks for reporting, we will work on a fix.

**Comment by leonid-s-usov** [comment]
We had to downgrade to 1.3 for now.

**Comment by mansbernhardt** [comment]
I think we will be able to have a new version out during the day, so if you could wait it would be great.

**Comment by leonid-s-usov** [comment]
That's cool but we have a new installation today so it's too late. We will issue an update once we verify the new version of the SDK.
Thanks.

**Comment by Sajjon** [comment]
@leonid-s-usov Hey Leonid, Alex from iZettle here. Can I just ask you real quick if you installed version 1.4 via Cocoapods or not?

**Comment by leonid-s-usov** [comment]
Hi Alex, yes we have installed it via Cocoapods, by running `pod update iZettleSDK`

**Comment by Sajjon** [comment]
@leonid-s-usov Thanks for your quick reply. 

I am just gonna check one thing real quick, let me get back to you in 10 minutes!

**Comment by Sajjon** [comment]
We are optimising our CI for making SDK releases, moving over to [fastlane](https://github.com/fastlane/fastlane), while testing our scripts an incorrect (premature) tag named `1.4` (as opposed to the correct tag with the name `1.4.0`) was created.

Out pod spec is pointing to `1.4.0`, so pods should not have chosen the incorrect `1.4`. 

A couple of hours ago I removed the incorrect tag `1.4`. 

So there will probably not be any change, but I am just curious, if you run:
`pod update`
`pod installl`

And try again, you are still experiencing the same crash right?

**Comment by Sajjon** [comment]
@leonid-s-usov We are unable to reproduce this crash using version `1.4.0`, the current theory we have is that the unwanted, incorrect git tag `1.4` contains code that causes this crash and that you might somehow have gotten it (although not refered to in `.podspec` (which is referring to `1.4.0`))

Please let me know if you are experiencing the same crash after:
`pod update`
`pod install`

**Comment by Sajjon** [comment]
@leonid-s-usov You might also try downloading the SDK manually [from releases here](https://github.com/iZettle/sdk-ios/releases/tag/1.4.0), instead of using Cocoapods. 

Please let me know how the results. Thank you very much!

**Comment by leonid-s-usov** [comment]
OK, I will try and update you afterwards. Thanks

**Comment by leonid-s-usov** [comment]
Just for your reference, this is the log from my original commit when updating to the new iZettle:

`Podfile.lock`

```
-  - iZettleSDK (1.3.0)
+  - iZettleSDK (1.4.0)

-  iZettleSDK: c192ec6ea6cf3a2c923ddf5191d3882f929a151f
+  iZettleSDK: 16cd6429791cd5acf00e7a8c2e598ab4cb68c82e
```

Now when calling `pod update iZettleSDK` I am getting the same commit.

**Comment by leonid-s-usov** [comment]
Here is the head of the CHANGES file

```
1.4.0 - 2017-01-24
- testing fastlane sdk build

v1.4.0 - 2016-01-23
- Partial refunds
- Stability and improvments

```

**Comment by Sajjon** [comment]
@leonid-s-usov We are soon rolling out `1.4.1` but maybe you can beta test it? In your Podfile, can you specify the branch named `fastlane` (as in build by our fastlane script):
https://github.com/iZettle/sdk-ios/tree/fastlane

So in the podfile you should write:
`pod 'iZettleSDK', :git => "https://github.com/iZettle/sdk-ios.git", :branch => "fastlane"`

This hope that this branch contains a fix for the problem.

Please let me know if it worked, then we will make a role out.

**Comment by leonid-s-usov** [comment]
It says 1.4.2 there on your fastlane branch..

**Comment by Sajjon** [comment]
@leonid-s-usov Thanks, yes, it got double bumped somehow, I will fix that today. But have you tried if it solved the problem?

**Comment by leonid-s-usov** [comment]
About to do that today

**Comment by Sajjon** [comment]
@leonid-s-usov We are going to release version `1.4.1` today. I fixed the incorrect naming and deleted the branch named `fastlane`, until we release `1.4.1` to pods please use this branch:
`sdk_version_1.4.1`
https://github.com/iZettle/sdk-ios/tree/sdk_version_1.4.1

**Comment by leonid-s-usov** [comment]
hm.. I have already made a build for QA using the head of the fastlane branch. Were there many changes?

**Comment by Sajjon** [comment]
@leonid-s-usov No same code, just fixed incorrect version name (to `1.4.1` instead of `1.4.2`) and correct dates for release notes, since I apparently registered that it is 2017 and not 2016 😂

So source code is the same. you can use the build based on the branch `fastlane` for QA 👍

**Comment by mansbernhardt** [comment]
The only change from the 1.4.2 build is the the version has been changed to 1.4.1.. But I your really should use the 1.4.1 final build when before shipping or we will have confusion down the road when releasing a 1.4.2

**Comment by Sajjon** [comment]
@leonid-s-usov Yeah as my colleague Måns said, it is of course better to use 1.4.1 so that there wont be any confusion with future releases.

So sorry for the inconvenience!

**Comment by leonid-s-usov** [comment]
I know, and I assumed that I would force update to the public 1.4.1 release once it was out. I will proceed with the QA of the former `fastlane` branch until you guys release 1.4.1 publicly.
Thanks

**Comment by leonid-s-usov** [comment]
For now there are no crashes with the low battery iZettle device. Looking forward to official 1.4.1 release

**Comment by mansbernhardt** [comment]
@leonid-s-usov Great! We are still confused why you had the crash in the first place, unless you or the podspec accidentally downloaded the pre-mature "1.4" build instead of the "1.4.0" build.

@leonid-s-usov Also let me know how the global enable/disable animation fix as well as the hasActiveSession worked out for you.

**Comment by Sajjon** [comment]
@leonid-s-usov  I just release version 1.4.1 (https://github.com/iZettle/sdk-ios/releases/tag/1.4.1)

It is available using cocoapods as well.

A big thanks for your patience and so sorry for the inconvenience. 

Can I close this issue?

**Comment by leonid-s-usov** [comment]
thanks, updating the source. I will close once the 1.4.1 doesn't crash, if that's ok ;)

Above I noted the spec checksum as it was available when updating to 1.4.0. You can use it to track exactly the version of the spec file that was available on cocoapods at that time

**Comment by Sajjon** [comment]
@leonid-s-usov thanks for the comment regarding spec checksum! 👍
